conf/modules.config

```diff
@@ -244,13 +265,65 @@ process {
         ]
     }
 
-    withName: SAMTOOLS_FLAGSTAT {
+    withName: ".*MAP:SAMTOOLS_FLAGSTAT" {
+        ext.prefix = { "${meta.id}_${meta.library_id}" }
         publishDir = [
             path: { "${params.outdir}/mapping/" },
             mode: params.publish_dir_mode,
             pattern: '*.flagstat'
         ]
     }
 
-}
+    withName: ".*DEDUPLICATE:PICARD_MARKDUPLICATES" {
+        ext.prefix = { "${meta.id}_${meta.library_id}_${meta.genomic_region}.MarkDuplicates" }
+        ext.args = [
+            "--REMOVE_DUPLICATES",
+            "--VALIDATION_STRINGENCY SILENT",
+            "--ASSUME_SORTED"
+        ].join(' ').trim()
+        publishDir = [
+            enabled: false
+        ]
+    }
+
+    withName: ".*DEDUPLICATE:DEDUP" {
+        ext.args = { params.dedup_all_merged ? '-m' : '' }
```

Comment:
Reason why not applying scatter/gather here? If we do want to do this (and I assume we do), need an updated prefix